Update About page to display correct Freetube logo based on currently set theme

ducks opened the pull request:
# Update About page to display correct Freetube logo based on currently set theme

## Pull Request Type
- [ ] Bugfix
- [x] Feature Implementation
- [ ] Documentation
- [ ] Other

## Related issue
closes #4020 

## Description
This PR updates the About page view to display the Freetube logo similarly to the top nav. It adds two HTML elements in place of the inline svg. Then we can use CSS to style those elements with the logo variables set by each theme.

## Screenshots <!-- If appropriate -->
before:
![before-nordic](https://github.com/FreeTubeApp/FreeTube/assets/868959/821dee6e-e6c7-400e-9fb6-4d467b697669)
![before-pastel-pink](https://github.com/FreeTubeApp/FreeTube/assets/868959/c9710e35-a98f-4511-a97c-0a2dfc05d367)

after:
![after-nordic](https://github.com/FreeTubeApp/FreeTube/assets/868959/1d9bb3a0-e2e7-47c2-b450-8c193fe68286)
![after-pastel-pink](https://github.com/FreeTubeApp/FreeTube/assets/868959/5f44c56d-130e-47cb-b475-9eeb1c50ac13)

## Desktop
<!-- Please complete the following information-->
- **OS: linux**
- **OS Version: manjaro 24**
- **FreeTube version: v0.20.0 beta**

absidue commented on src/renderer/views/About/About.scss:
```diff
@@ -62,3 +57,30 @@
+  justify-content: center;
+  inline-size: 100%;
+
+  .logoIcon {
+    background-image: var(--logo-icon);
+    background-position: right top;
+    background-repeat: no-repeat;
+    background-size: 200px;
+    block-size: 200px;
+    inline-size: 200px;
+  }
+
+  .logoText {
+    background-image: var(--logo-text);
+    background-position: right top;
+    background-repeat: no-repeat;
+    background-size: 350px;
+    block-size: 175px;
+    position: relative;
+    inline-size: 350px;
+  }
+}
```
We don't need nesting for any of these classes, as Vue's scoped CSS feature (activated by the `scoped` attribute on the `<style>` tag) already takes care of making sure it only applies to this component.

Yes, I am aware that we have a bunch of unnecessary nesting in other files in the project, but I do eventually want to get rid of it, as it increases the size of the CSS output without providing any benefit.

```suggestion
}

.logoIcon {
  background-image: var(--logo-icon);
  background-position: right top;
  background-repeat: no-repeat;
  background-size: 200px;
  block-size: 200px;
  inline-size: 200px;
}

.logoText {
  background-image: var(--logo-text);
  background-position: right top;
  background-repeat: no-repeat;
  background-size: 350px;
  block-size: 175px;
  position: relative;
  inline-size: 350px;
}
```

ducks replied:
updated to remove nesting

absidue reviewed (commented):
Thank you for you pull request! I haven't had a chance to test it yet, so for the moment this is just a code review with a small code cleanup suggestion.

absidue commented:
Could you please use relative sizes instead of absolute pixel amounts, as the logo is larger than it was before on smaller screen sizes and on mobile is gigantic and split onto two separate lines. I've provided some before and after screenshots below demonstrating the issue:

<details><summary>Desktop before</summary>

![before](https://github.com/FreeTubeApp/FreeTube/assets/48293849/141808bc-3a1b-487b-a9c3-41cf335fad8e)

</details>

<details><summary>Desktop with this pull request</summary>

![after](https://github.com/FreeTubeApp/FreeTube/assets/48293849/c21480f1-9aef-4ffa-a3fd-34b5cef48dd9)

</details>
<details><summary>Mobile before</summary>

![before-mobile](https://github.com/FreeTubeApp/FreeTube/assets/48293849/85dde2b9-bc46-447b-9764-ab23cf587808)

</details>
<details><summary>Mobile with this pull request</summary>

![after-mobile](https://github.com/FreeTubeApp/FreeTube/assets/48293849/cef16619-8a51-45c1-ac8c-cb1c1ad42b60)

</details>

ducks commented:
sure, np. I started with rems locally but saw some px values and decided to just stay consistent to start.

I added some css to specifically break the Freetube text to a newline when it gets smaller but I can remove that. The top nav just hides the logo text and shows only the icon below 680px. should I replicate that here?

absidue commented:
> The top nav just hides the logo text and shows only the icon below 680px. should I replicate that here?

We have more space on the About page, so ideally it should like the same as it was before, just in different colors.

ducks commented:
okie doke. I will work on making it match what was there size and layout wise. I mostly ask because before, the about page just had a single inline svg that scaled down but this PR adds the two elements that get styled so it could be more flexible if desired.

ducks commented:
I ended up pivoting on how to implement this and added an ft-logo-full component. this way, the logo can be imported directly as an svg and then styled with css using fill. I added classes to the svg and then a few new logo variables to each theme that will style the svg when enabled.

I named it ft-logo-full because this could be done for the separate icon and text parts in the top nav and could eliminate the need for logo files for each theme.

<details>
  <summary>mobile before and after:</summary>

![ft-before-mobile](https://github.com/FreeTubeApp/FreeTube/assets/868959/b9a395f0-0635-480e-9df9-56725d520bb1)

![ft-after-mobile](https://github.com/FreeTubeApp/FreeTube/assets/868959/125ab295-8e70-47b3-9690-d7d67139739f)
</details>

<details>
  <summary>tablet before and after:</summary>

![ft-before-tablet](https://github.com/FreeTubeApp/FreeTube/assets/868959/9f726692-760d-4462-b65a-c6803be93e27)


![ft-after-tablet](https://github.com/FreeTubeApp/FreeTube/assets/868959/df0af65c-080d-4500-9e87-dbf0f1435051)
</details>

<details>
  <summary>desktop before and after:</summary>

![ft-before-desktop](https://github.com/FreeTubeApp/FreeTube/assets/868959/18f61b19-ed78-4472-9313-052d0847d9b3)

![ft-after-desktop](https://github.com/FreeTubeApp/FreeTube/assets/868959/4b001776-48bb-4402-adf6-6a87955ab5d5)

</details>

ducks commented on src/renderer/themes.css:
```diff
@@ -170,6 +173,9 @@
   --search-bar-color: #3E3F4A;
   --logo-icon: url("../../_icons/iconDraculaLightSmall.svg");
   --logo-text: url("../../_icons/textDraculaLightSmall.svg");
+  --logo-primary-color: #f8f8f8;
```
ah yeah, that's a good idea. I can do that before merging.

ducks replied:
pushed an update that sets the three logo color variables to the primary-text-color variable for themes that have a logo with only a single color. I left the css that sets the specific colors for the dark, light, and black themes.

jasonhenriquez reviewed (approved):
Works as intended. Not sure if there's a way to do this while keeping the SVG as a file using attribute selectors, but I haven't been able to find one.

ducks commented:
> Works as intended. Not sure if there's a way to do this while keeping the SVG as a file using attribute selectors, but I haven't been able to find one.



> Works as intended. Not sure if there's a way to do this while keeping the SVG as a file using attribute selectors, but I haven't been able to find one.

@jasonhenriquez do you mean doing this without a component and just the svg file?

ducks commented:
> Or just a reference to the file rather than having it in a template. I don't think so, from what I've seen, so what's present now should be fine.

ah yeah. It is possible to import the svg in the js and include it as a component that way, but that would require vue-svg-loader or similar.
